x-pack/plugins/apm/public/hooks/useFetcher.tsx

```diff
@@ -27,21 +27,28 @@ export function useFetcher<Opts, Response>(
     error?: Error;
   }>({});
 
-  const useEffectKey = [...Object.keys(options), ...Object.values(options)];
+  const useEffectKey = [...Object.keys(fnOptions), ...Object.values(fnOptions)];
```

Comment:
On the other hand, allowing people to forget to use `useMemo` and end up with a fetcher that refetches constantly is probably bad.

Reply:
Currently the API is:
```js
const { data } = useFetcher(fn, fnArgs);
```

Where `fnArgs` are both applied to `fn`, and used as key. Do you suggest something like?
```js
const { data } = useFetcher(fn, fnArgs, key);
```

In a real life scenario that would look like:
```js
const { data: errorDistributionData } = useFetcher(
  loadErrorDistribution,
  { serviceName, start, end },
  [serviceName, start, end]
);
```

Looking at it, it's not bad actually. It feels a bit redundant but maybe that's a good thing. That would actually allow us to avoid the separation of function and args:
```
const { data: errorDistributionData } = useFetcher(
  () => loadErrorDistribution({ serviceName, start, end }),
  [serviceName, start, end]
);
```

That API is actually identical to `useMemo` - I think I very much prefer that suddenly :)

Reply:
Much less magic. More standard. If this is what you meant, I'll go ahead and make the changes.

Reply:
I was thinking of having the user call `useMemo` on the options object before passing it to `useFetcher` but I didn't like that idea. What you have is great, much clearer and probably more lint-able in the future. A+